custom_components/rivian/__init__.py

```diff
@@ -159,9 +159,14 @@ async def _update_api_data(self):
                 properties=sensors,
             )
             vijson = await vehicle_info.json()
+            _LOGGER.debug(vijson)
 
             vehicle_info_items = self.build_vehicle_info_dict(vijson)
-            return vehicle_info_items
+            if vehicle_info_items:
+                self._previous_vehicle_info_items = vehicle_info_items
+                return vehicle_info_items
+            else:
+                return self._previous_vehicle_info_items
```

Comment:
If we don't have the unavailable info in the data of the sensors, we should log these events so we know when it happens.

Reply:
Good point and maybe dump what did come back into logs for future debugging?

Reply:
What happens if it goes unavailable for 10 minutes (arbitrary time). Will it always use the old value?   

Should we set a, if unavailable 10 times, show as unavailable?

Reply:
to sum up the discord chat, this `else` statement is only triggered when the entire call to Rivian fails to return **any** sensor data vs simply omitting a single sensor data point.